cmake/module/FindBerkeleyDB.cmake

```diff
@@ -2,86 +2,130 @@
 # Distributed under the MIT software license, see the accompanying
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-if(CMAKE_HOST_APPLE)
+#[=======================================================================[
+FindBerkeleyDB
+--------------
+
+Finds the Berkeley DB headers and library.
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+This module provides imported target ``BerkeleyDB::BerkeleyDB``, if
+Berkeley DB has been found.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This module defines the following variables:
+
+``BerkeleyDB_FOUND``
+  "True" if Berkeley DB found.
+
+``BerkeleyDB_VERSION``
+  The MAJOR.MINOR version of Berkeley DB found.
+
+#]=======================================================================]
+
+if(BREW_COMMAND)
+  # The Homebrew package manager installs the berkeley-db* packages as
+  # "keg-only", which means they are not symlinked into the default prefix.
+  # To find such a package, the find_path() and find_library() commands
+  # need additional path hints that are computed by Homebrew itself.
   execute_process(
-    COMMAND brew --prefix berkeley-db@4
-    OUTPUT_VARIABLE bdb4_brew_prefix
+    COMMAND ${BREW_COMMAND} --prefix berkeley-db@4
+    OUTPUT_VARIABLE _BerkeleyDB_homebrew_prefix
     ERROR_QUIET
     OUTPUT_STRIP_TRAILING_WHITESPACE
   )
+  set(_BerkeleyDB_homebrew_include_hint "${_BerkeleyDB_homebrew_prefix}/include")
```

Comment:
Can the output variable of this not just be `_BerkeleyDB_homebrew_include_hint`, and then we avoid the intermediate variable and the `set` / `unset` etc. All this does is append /include?

Reply:
Thanks! Reworked.